lib/readline.js

```diff
@@ -500,6 +518,60 @@ Interface.prototype._insertString = function(c) {
     // a hack to get the line refreshed if it's needed
     this._moveCursor(0);
   }
+
+  // searchHistory.call(this);
+  this._searchHistory();
+};
+
+Interface.prototype._appendSearchResult = function(result) {
```

Comment:
I would rather not add more underscore properties (also adding new properties on the instance). Instead, you could just use a function that receives the actual interface instance as first argument. That applies to all added functions. Everything accessible is "public" afterwards.

Reply:
@BridgeAR Are there any benefits to this approach? I see several functions like `_insertString` etc, are already implemented as _ properties and hence I followed the same approach here as well.

Reply:
Yes, it is not exposed to the users if it's not on the prototype. If something is accessible to users, there is a high likelihood that these properties will be monkey patched, used in ways they were not meant to be used and more. That prevents any changes to these internal functions without a full deprecation cycle.

There was already ongoing effort to do the opposite: to try to make more functions internal without exposing them to the users. @lance did some of that work.